### ISSUE TITLE
Star Scanned no luminosityclass in body and problem with rings

### ISSUE THREAD

**Issue by StivlWrith** [EDCD/EDDI#133]
I make some scans and I see that when there is a scan on stars, the event.luminosityClass is not in the reportbody
I make some changes in the code, ans, because I think I will never understand how to use correctly GitHub, i put the changes in txt here (I use your develop branch for that).

I make also some changes to have the rings appear also with stars (I know that the ReserveLevel don't appear actually with star, but it will be here if one day FD add it) and also to have event.luminosityclass to replace event.luminosityClass .

[Body.cs.txt](https://github.com/EDCD/EDDI/files/1352128/Body.cs.txt)
[EDDI.cs.txt](https://github.com/EDCD/EDDI/files/1352131/EDDI.cs.txt)
[JournalMonitor.cs.txt](https://github.com/EDCD/EDDI/files/1352132/JournalMonitor.cs.txt)
[StarClass.cs.txt](https://github.com/EDCD/EDDI/files/1352129/StarClass.cs.txt)
[StarScannedEvent.cs.txt](https://github.com/EDCD/EDDI/files/1352130/StarScannedEvent.cs.txt)

**Comment by Tkael** [comment]
Not surprising - jgm hasn't updated the server to capture luminosityclass. When we transition the server, that should resolve this issue. Changes to the code should go into a pull request... it's difficult for us to evaluate / compare them and see what's changed in text files.

**Comment by Darkcyde13** [comment]
This is one thing I had in my Star Scanned  and Star Report scripts for testing, and they have both worked perfectly for me since 2.4 was released.  I have a section in Star Report for this:
```
    {if reportbody.luminosityclass:
        with a luminosity class of {reportbody.luminosityclass},
    }
```
This reports the class as I would expect it, with things like 'V' or 'Va'.

> 
>  Scan of M-class star completed. Star C is a very common main sequence star with a luminosity class of Va, that has been around for over 10 billion years. As you can see, it emits a orange-red light.
>  Scan of K-class star completed. Star A is a common main sequence star with a luminosity class of V, that is over 10 billion years old. As you can see, it emits a yellow-orange light.

Note that I have trimmed my script so that stars that contain the system name as part of their name are noly reffered to as 'A', 'B', 'C', etc.

**Comment by Tkael** [comment]
@Darkcyde13 Your script must be picking it up from the event then. It's not present on the server. See for example: http://api.eddp.co/systems/HD%2043193. There is no luminosity class data included. 

The test code below also gives no luminosityclass:
```{set reportbody to BodyDetails("Sol", "Sol")}
For {reportbody.name}, the luminosity class is {reportbody.luminosityclass}.```

**Comment by Darkcyde13** [comment]
My apologies.  When StivlWrith said:- 

> I make some scans and I see that when there is a scan on stars, the event.luminosityClass is not in the reportbody

I read that as meaning event.luminosityclass wasn't working.  I've since realised that I made a change to the default Star Report script to have reportbody set from the event, and failing that, two fallback checks to set from BodyDetails() instead, in case the script wasn't being called from the event itself.  And it's that part I hadn't got around to testing yet.
